reposcan/database/modules_store.py

```diff
@@ -2,14 +2,33 @@
 Module containing classes for fetching/importing modules from/into database.
 """
 
+import re
+
 from psycopg2.extras import execute_values
 
 from database.object_store import ObjectStore
 
-
 class ModulesStore(ObjectStore):
     """Class providing interface for storing modules and related info."""
 
+    def __init__(self):
+        super().__init__()
+        self.nevra_re = re.compile(r'(.*)-(([0-9]+):)?([^-]+)-([^-]+)\.([a-z0-9_]+)')
+
+    def _split_packagename(self, filename):
```

Comment:
We have function for that already. It's `common.rpm.parse_rpm_name`. Could you use it instead of this, please?

Reply:
common.rpm.parse_rpm_name doesn't work identically to _split_packagename.  For example, it throws RPMParseException for 'perl-DBD-Pg-0:3.7.4-2.module+el8+2517+b1471f1c.x86_64'

Reply:
It should parse it to:
('perl-DBD-Pg', '0', '3.7.4', '2.module+el8+2517+b1471f1c', 'x86_64')

Reply:
On closer inspection, the _split_packagename() and common.rpm.parse_rpm_name() methods are different.  It looks like common.rpm.parse_rpm_name() doesn't correctly parse out the epoch when passed "perl-DBD-Pg-0:3.7.4-2.module+el8+2517+b1471f1c.x86_64.rpm".
It returns ('perl-DBD-Pg', None, '0:3.7.4', '2.module+el8+2517+b1471f1c', 'x86_64')

I'm going to leave the _split_packagename() method in place for now.  I think we need to open a bug on the parse_rpm_name() method and also improve it so it works with a packagename which doesn't have the trailing ".rpm".  Possibly it should just be replaced with the code that implements _split_packagename().  But first we should check that it is actually causing problems where it is being used.